### PR TITLE
[trainer] identify skipped tests in `xcresult` and export to Junit format

### PR DESCRIPTION
**Pull request by igor-makarov**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

File: trainer/lib/trainer/test_parser.rb [hunk 1]
```diff
@@ -319,6 +324,7 @@ def summaries_to_data(summaries, failures, output_remove_retry_attempts: false)
         # Used for seeing if any tests continued to fail after all of the Xcode 13 (and up) retries have finished
         unique_tests = tests_by_identifier.values || []
         row[:number_of_tests_excluding_retries] = unique_tests.count
+        row[:number_of_skipped] = unique_tests.map { |a| a[:skip_count] }.inject(:+)
         row[:number_of_failures_excluding_retries] = unique_tests.find_all { |a| a[:success_count] == 0 }.count
```
Comment (joshdholtz) [inline trainer/lib/trainer/test_parser.rb:328]:
```suggestion
        row[:number_of_failures_excluding_retries] = unique_tests.find_all { |a| a[:success_count] == 0 && a[:skip_count] == 0 }.count
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
The failure count does not take into consideration if a test has been skipped so I suggested a fix for that!

Can we also get the test result output table in `scan` to show number skipped? 😇 

And thanks for making this PR!

**Comment by igor-makarov** [comment]
Hey Josh! I've accepted your CR suggestion (with a minor variation to keep linter happy)!

I've also added the summary row to `scan` output but - I haven't had the chance to test it with a real-life test run and I wasn't able to figure out where to add testing for it in the runner specs.

**Comment by joshdholtz** [comment]
This is getting close! It looks like `:number_of_skipped` needs to be added in these two places yet 👇 

Screen Shot 2022-02-01 at 8 52 10 PM
Screen Shot 2022-02-01 at 8 52 22 PM

**Comment by joshdholtz** [comment]
And side note: I'm not able to add commits from a fork from an organization 😔 I don't know why GitHub doesn't allow that. In the future you want to make a fork from your personal account I can help add some commits!

**Pull request closed without merging**